### ISSUE TITLE
cgroup module: socket rules per cgroup, no boot change

### ISSUE THREAD

**Issue by lneto** [luainkernel/lunatik#666]
Part of the LSM epic (#658). First phase, and deliberately the one that needs no boot change.

## Context

Cgroup BPF programs gate socket operations per cgroup: `connect`, `bind`, `sendmsg` and their v6
twins, plus device access and sockopt. They need only `CONFIG_CGROUP_BPF`, they attach with
`bpftool cgroup attach` with no custom loader, and they work on a stock distribution kernel.

That makes this the phase that proves the whole path — stub, kfunc, ctx, verdict, tests — before
anything depends on a reboot.

Kfunc reachability is verified: `BPF_PROG_TYPE_CGROUP_SOCK_ADDR` maps to `BTF_KFUNC_HOOK_CGROUP_SKB`
(`kernel/bpf/btf.c:7910`, Linux 6.8), so a kfunc registered for that program type is callable from a
`cgroup/connect4` program.

## Scope

A new module, `cgroup` (`lib/luacgroup.c`), publishing `bpf_luacgroup_run` and a ctx object:

    local cgroup = require("cgroup")
    local errno  = require("linux.errno")

    local function guard(ctx)
        if ctx:address() == BLOCKED then
            return -errno.EPERM
        end
        return 0
    end

    cgroup.attach(guard)

and the stub, attached with `bpftool cgroup attach /sys/fs/cgroup/<path> cgroup_inet4_connect`:

    SEC("cgroup/connect4")
    int connect4(struct bpf_sock_addr *ctx)
    {
            return bpf_luacgroup_run(rt_key, sizeof(rt_key), ctx, NULL, 0) == 0;
    }

Note the inversion on that last line: a cgroup socket program returns **1 to allow and 0 to deny**,
the opposite of the LSM convention. Lua speaks one language — `0` allows, negative errno denies — and
the stub translates. Keep the translation in the stub so every callback in this project reads the same
way.

The ctx exposes `family`, `protocol`, `address`, `port` and `argument`. `address` and `port` are
settable: `connect4` can redirect as well as refuse, and hiding that would make the binding weaker
than the program it wraps.

## Kernel APIs

    register_btf_kfunc_id_set(BPF_PROG_TYPE_CGROUP_SOCK_ADDR, &set)   EXPORT_SYMBOL_GPL

Follow `lib/luaxdp.c` for the pattern and the version guards, or `lunatik_ebpf.h` if phase 0 decided
that way. `sudo make btf_install` must run for the stub to resolve the kfunc; a program that cannot
see it fails to load with "kernel function ... not found".

`attach` requires a non-sleepable runtime (`lunatik run <script> softirq`), like `xdp.attach`: the
kfunc is called from a program that cannot sleep.

## Files touched

`lib/luacgroup.c`, `lib/Kbuild`, `Kconfig`, `config.ld`, `README.md`, `examples/netguard/` (stub plus
Lua script plus Makefile, following `examples/filter/`), `tests/cgroup/`, `tests/README.md`.

## Acceptance criteria

- [ ] `cgroup.attach` from a non-sleepable runtime registers; from a sleepable one it raises
- [ ] a callback returning 0 lets the connection through
- [ ] a callback returning `-EPERM` makes `connect` fail inside the test cgroup, while the same
      connection succeeds outside it
- [ ] `ctx:address()` and `ctx:port()` match what the test dialled
- [ ] setting `ctx:address()` redirects the connection to a local listener
- [ ] data packed by the stub reaches Lua intact and unpacks with `struct`
- [ ] `cgroup.detach()` restores unfiltered traffic
- [ ] a stub naming a runtime that does not exist fails loudly rather than silently allowing
- [ ] tests create their own cgroup, move only a subshell into it, and detach in a `trap`
- [ ] LDoc, `config.ld` in alphabetical order, README row, `tests/README.md`

## Notes

Never attach to the root cgroup while developing. A deny rule there takes the machine off the network,
including whatever you were using to watch the test.

Effort: medium.
